pkg/install/generate.go

```diff
@@ -167,248 +139,90 @@ func InstallerPodSpec(
-					SecretName: cd.Spec.Platform.Azure.CredentialsSecretRef.Name,
-				},
-			},
-		})
-		volumeMounts = append(volumeMounts, corev1.VolumeMount{
-			Name:      "azure",
-			MountPath: azureAuthDir,
-		})
-		env = append(env, corev1.EnvVar{
-			Name:  "AZURE_AUTH_LOCATION",
-			Value: azureAuthFile,
-		})
+		credentialRef = cd.Spec.Platform.Azure.CredentialsSecretRef.Name
+		emptyDirs["azure"] = constants.AzureCredentialsDir
 	case cd.Spec.Platform.GCP != nil:
-		volumes = append(volumes, corev1.Volume{
-			Name: "gcp",
-			VolumeSource: corev1.VolumeSource{
-				Secret: &corev1.SecretVolumeSource{
-					SecretName: cd.Spec.Platform.GCP.CredentialsSecretRef.Name,
-				},
-			},
-		})
-		volumeMounts = append(volumeMounts, corev1.VolumeMount{
-			Name:      "gcp",
-			MountPath: gcpAuthDir,
-		})
-		env = append(env, corev1.EnvVar{
-			Name:  "GOOGLE_CREDENTIALS",
-			Value: gcpAuthFile,
-		})
+		credentialRef = cd.Spec.Platform.GCP.CredentialsSecretRef.Name
+		emptyDirs["gcp"] = constants.GCPCredentialsDir
 	case cd.Spec.Platform.OpenStack != nil:
-		volumes = append(volumes, corev1.Volume{
-			Name: "openstack",
-			VolumeSource: corev1.VolumeSource{
-				Secret: &corev1.SecretVolumeSource{
-					SecretName: cd.Spec.Platform.OpenStack.CredentialsSecretRef.Name,
-				},
-			},
-		})
-		volumeMounts = append(volumeMounts, corev1.VolumeMount{
-			Name:      "openstack",
-			MountPath: openStackCloudsDir,
-		})
+		credentialRef = cd.Spec.Platform.OpenStack.CredentialsSecretRef.Name
+		emptyDirs["openstack"] = constants.OpenStackCredentialsDir
 		if cd.Spec.Platform.OpenStack.CertificatesSecretRef != nil && cd.Spec.Platform.OpenStack.CertificatesSecretRef.Name != "" {
-			volumes = append(volumes, corev1.Volume{
-				Name: "openstack-certificates",
-				VolumeSource: corev1.VolumeSource{
-					Secret: &corev1.SecretVolumeSource{
-						SecretName: cd.Spec.Platform.OpenStack.CertificatesSecretRef.Name,
-					},
-				},
-			})
-			volumeMounts = append(volumeMounts, corev1.VolumeMount{
-				Name:      "openstack-certificates",
-				MountPath: openStackCADir,
-			})
+			emptyDirs["openstack-certificates"] = constants.OpenStackCertificatesDir
+			certificateRef = cd.Spec.Platform.OpenStack.CertificatesSecretRef.Name
 		}
 	case cd.Spec.Platform.VSphere != nil:
-		volumes = append(volumes, corev1.Volume{
-			Name: "vsphere-certificates",
-			VolumeSource: corev1.VolumeSource{
-				Secret: &corev1.SecretVolumeSource{
-					SecretName: cd.Spec.Platform.VSphere.CertificatesSecretRef.Name,
-				},
-			},
-		})
-		volumeMounts = append(volumeMounts, corev1.VolumeMount{
-			Name:      "vsphere-certificates",
-			MountPath: vsphereCloudsDir,
-		})
-		env = append(env, vSphereCredsEnvVars(cd.Spec.Platform.VSphere.CredentialsSecretRef.Name)...)
+		credentialRef, certificateRef = cd.Spec.Platform.VSphere.CredentialsSecretRef.Name, cd.Spec.Platform.VSphere.CertificatesSecretRef.Name
+		// TODO: I don't think we actually use this dir. Can we get rid of it?
```

Comment:
IIRC we use environment variables for the credentials and we can probably remove this.

Reply:
Ack. Prefer to do it in a fup jic.